packages/block-library/src/utils/get-transformed-attributes.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * WordPress dependencies
+ */
+import { getBlockType, hasBlockSupport } from '@wordpress/blocks';
+
+/**
+ * Transform block support attributes and metadata during block transforms.
+ *
+ * @param {Object}   attributes       Original attributes from the block being transformed.
+ * @param {string}   newBlockName     Name of the target block after transformation.
+ * @param {Function} bindingsCallback Optional callback to transform the `bindings` property object.
+ * @return {Object} New attributes object with preserved block support attributes and metadata.
+ */
+export function getTransformedAttributes(
+	attributes,
+	newBlockName,
+	bindingsCallback = null
+) {
+	if ( ! attributes ) {
+		return undefined;
+	}
+
+	const newBlockType = getBlockType( newBlockName );
+	if ( ! newBlockType ) {
+		return undefined;
+	}
+
+	const transformedAttributes = {};
+
+	// Handle attributes derived from block support.
+	if (
+		hasBlockSupport( newBlockType, 'allowedBlocks' ) &&
+		attributes.allowedBlocks
+	) {
+		transformedAttributes.allowedBlocks = attributes.allowedBlocks;
+	}
+	if ( hasBlockSupport( newBlockType, 'anchor' ) && attributes.id ) {
```

Comment:
This should be `attributes.anchor`, no?

Reply:
Hm.. Not very experienced with these kind of attributes, but it seems this has to do with the parsing/rendering of the block, but the actual block attribute is [anchor](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/hooks/anchor.js#L47).